riocli/device/model.py

```diff
@@ -25,6 +27,10 @@ def __init__(self, *args, **kwargs):
         self.update(*args, **kwargs)
 
     def find_object(self, client: Client) -> bool:
+        # For virtual devices, always return False to ensure re-onboarding is done even if the device already exists
+        if self.spec.get('virtual', {}).get('enabled', False):
+            return False
```

Comment:
On a second thought, let's implement `update_object` and `delete_object` and get rid of this condition in a separate PR.

The update_object implementation can take care of re-onboarding.